src/main/play-doc/general/MigrationGuide.md

```diff
@@ -33,18 +33,425 @@ In additional to disabling Elasticsearch via configuration, a service lookup mus
+
+Populate the "Custom Endpoint Configuration" section described in the [[populate custom HAProxy configuration template|DynamicProxyConfiguration#Populate-the-custom-HAProxy-configuration-template]] step as such:
+
+```
+{{#eachAcls bundles defaultHttpPort=9443}}
+  {{#ifAcl 'my-bundle' '1.1' 'endpoint-label'}}
+frontend my_www_frontend
+  bind {{haproxyHost}}:6789
+  mode http
+  acl my_www_path_match path_beg /my-service
+  use_backend my_www_backend if my_www_path_match
+
+backend my_www_backend
+  mode http
+  reqrep ^([^\ :]*)\ /my-service/?(.*) \1\ /\2
+    {{#eachBackendServer}}
+  server {{serverName}} {{host}}:{{port}} maxconn 1024
+    {{/eachBackendServer}}
+  {{/ifAcl}}
+{{/eachAcls}}
+
+```
+
+The snippet above will expose other HTTP endpoint at port `9443`, this value can be replaced with any valid port number accordingly.
+
+##### Host header match
+
+A service may declare an endpoint exposed on a particular host address. Given the service URI `http://www.acme.com`, the service is accessible from the outside world on HTTP port `80` when the HTTP host header is `www.acme.com`.
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+name := "my-bundle"
+BundleKeys.compatibilityVersion := "1.1"
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http://www.acme.com")))
+)
+
+```
+
+To:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, "my-service",
+    RequestAcl(Http("^/".r))
+  )
+)
+```
+
+Where `my-service` is the service name for the endpoint for the purpose of service lookup.
+
+###### Migrating bundle.conf
+
+Modify the `endpoints` section declared within the `bundle.conf` as such.
+
+From:
+
+```
+    system             = "my-bundle"
+    systemVersion      = "1.1"
+    endpoints          = {
+      "endpoint-label" = {
+        bind-protocol  = "http"
+        bind-port      = 0
+        services       = ["http://www.acme.com"]
+      }
+    }
+```
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "my-service"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+Where `my-service` is the service name for the endpoint for the purpose of service lookup.
+
+
+###### Supplying customized HAProxy configuration
+
+The [[Dynamic proxy configuration|DynamicProxyConfiguration]] describes the steps required to generate a customized HAProxy configuration.
+
+Populate the "Custom Endpoint Configuration" section described in the [[populate custom HAProxy configuration template|DynamicProxyConfiguration#Populate-the-custom-HAProxy-configuration-template]] step as such:
```

Comment:
@markusjura - added link to steps required to supply customized HAProxy configuration